Add Two EN Documents (#54)
OpenFunction/website: content/en/docs/best-practices/skywalking-solution-for-openfunction.md

---

> title: "Use SkyWalking for OpenFunction as a Distributed Tracing Solution"

Reviewer: ```suggestion
title: "Use SkyWalking for OpenFunction as an Observability Solution"
```

From https://github.com/apache/skywalking/pull/8789#discussion_r841063944

linkTitle: "Use SkyWalking for OpenFunction as a Distributed Tracing Solution"
weight: 5300
description: >	
    Learn how to use SkyWalking for OpenFunction as a distributed tracing solution.
---

This document describes how to use SkyWalking for OpenFunction as a distributed tracing solution.

## Overview

Although FaaS allows developers to focus on their business code without worrying about the underlying implementations, it is difficult to troubleshoot the service system. OpenFunction tries to introduce capabilities of observability to improve its usability and stability.

> SkyWalking provides solutions for observing and monitoring distributed systems in many different scenarios. OpenFunction has introduced SkyWalking (go2sky) as a distributed tracing solution for Go language functions.

Reviewer: ```suggestion
SkyWalking provides solutions for observing and monitoring distributed systems in many different scenarios. OpenFunction has bundled go2sky(SkyWalking's Golang agent) in OpenFunction tracer options to provide distributed tracing, statistics of function performance, and functions dependency map.
```

Author: [Updated](https://github.com/OpenFunction/website/pull/54/commits/ad145130cc8a8d23cec9acf1736aed2c187196d9).

## Prerequisites

- You have [installed OpenFunction](../../getting-started/installation/).
- You have [created a secret](../../getting-started/your-first-function/function-go/#create-a-secret).

> - You have [installed SkyWalking](https://github.com/apache/skywalking-kubernetes#apache-skywalking-kubernetes).

Reviewer 2: It should be updated to `installed SkyWalking v9` since the FaaS support has just been introduced in SkyWalking v9, see [this comment](https://github.com/apache/skywalking/pull/8789#discussion_r841063853) for the context.

Author: [Updated](https://github.com/OpenFunction/website/pull/54/commits/1cf62c9e09b0cf4e62de025a5c4ad5d48f74dd30).

         name: "skywalking"
         oapServer: "skywalking-oap:11800"
       tags:
         func: tracing-function
         layer: faas
         tag1: value1
         tag2: value2
       baggage:
         key: "key1"
         value: "value1"
   ```

### Option 2: function-level configuration

To enable tracing configuration in the function-level, add the field `plugins.tracing` under `metadata.annotations` in the function manifest as the following example.

```yaml
metadata:
  name: tracing-function
  annotations:
    plugins.tracing: |
      enabled: true
      provider:
        name: "skywalking"
        oapServer: "skywalking-oap:11800"
      tags:
        func: tracing-function
        layer: faas
        tag1: value1
        tag2: value2
      baggage:
        key: "key1"
        value: "value1"
```

It is recommended that you use the global tracing configuration, or you have to add function-level tracing configuration for every function you create.

## Use SkyWalking as a Distributed Tracing Solution

1. Create functions by referring to [this document](../interact-with-dapr-output-binding/).

2. Then, you can observe the flow of entire link on the SkyWalking UI.

>    ![](/images/docs/en/best-practices/skywalking-solution-for-openfunction/tracing-topology.gif)

Reviewer 2: I think it would be better to use the SkyWalking v9 UI in the latest screenshots. It's not a big deal, it could be iterated later.

Author: OK.
